aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncDeltaSyncDBOperations.java

```diff
@@ -0,0 +1,162 @@
+
+/**
+ * Copyright 2018-2018 Amazon.com,
+ * Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Amazon Software License (the "License").
+ * You may not use this file except in compliance with the
+ * License. A copy of the License is located at
+ *
+ *     http://aws.amazon.com/asl/
+ *
+ * or in the "license" file accompanying this file. This file is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, express or implied. See the License
+ * for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.appsync;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.database.sqlite.SQLiteStatement;
+
+final class AWSAppSyncDeltaSyncDBOperations {
+
+    final class DeltaSyncRecord {
+        long id;
```

Comment:
Tim made a comment on the iOS one, which is applicable here as well is to save some space by not using a separate id and key and switch to using only the key.

Reply:
Something to consider for later, right now it is tied in with the cancellation logic and will be a larger change.